### ISSUE TITLE
`sqlfluff fix` generates unparsable SQL

### ISSUE THREAD

**Issue by nakamichiworks** [sqlfluff/sqlfluff#4071]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.


### What Happened

Consider the following SQL (`file.sql`):

```sql
select x from "window";
```

Running `sqlfluff fix -f file.sql` fixes L059 violation and generates the SQL below:

```sql
select x from window;
```

However, linting this file fails due to parsing error.

```
❯ sqlfluff lint file.sql --dialect=ansi
== [file.sql] FAIL                                                                                                                                                                                                         
L:   1 | P:  10 |  PRS | Line 1, Position 10: Found unparsable section: 'from'
L:   1 | P:  14 |  PRS | Line 1, Position 14: Found unparsable section: ' window'
WARNING: Parsing errors found and dialect is set to 'ansi'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### Expected Behaviour

Generated SQL is valid and should be parsable.

### Observed Behaviour

Described above

### How to reproduce

Described above

### Dialect

`ansi` and `redshift`

### Version

sqlfluff 1.4.2 with Python 3.10.6

### Configuration

No config file

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by ayussh-verma** [comment]
Seems to be fixed now in `sqlfluff, version 3.2.5`
